### PR TITLE
Instrument editor now uses TGUI

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Instruments now use TGUI as their editor which is pretty cool.
It's mostly a 1:1 remake of the HTML UI except I did make a change to make the playback options a little more compact, leaving some more space for the editor before you have to scroll, and some other minor things that were made to make the UI hopefully nicer to look at and mess with.

When there's a song to play - While playing, Repeat section can't be edited
image

Help section and UI when there's no song put in
image

## Why It's Good For The Game

It is yet another step in finishing up https://hackmd.io/XLt5MoRvRxuhFbwtk4VAUA?view
Instruments especially were in a poor spot because they didn't respect things like ``IN_USE`` to not refresh if it's not the "UI" you are on, and such.

## Changelog

:cl:
refactor: Instruments now use TGUI.
/:cl:

### CODE REVIEW

**Comment by AnturK** [comment]
I know it's pretty basic conversion but could you make these inputs into actual inputs please, the button : value format is really ugly.

File: tgui/packages/tgui/interfaces/InstrumentEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        </Collapsible>
+        <Collapsible title="Help Section" icon="question">
+          <HelpSection />
+        </Collapsible>
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const InstrumentSettings = (props) => {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/InstrumentEditor.tsx:58]: are these being used elsewhere? if not, please remove the export

File: code/modules/instruments/items.dm [hunk 2]
```diff
@@ -201,19 +188,24 @@
...
 	. = ..()
-	UnregisterSignal(M, COMSIG_MOB_SAY)
+	UnregisterSignal(user, COMSIG_MOB_SAY)
+
+/obj/item/instrument/harmonica/proc/handle_speech(datum/source, list/speech_args)
+	SIGNAL_HANDLER
+	if(!song.playing)
+		return
+	if(!ismob(loc))
+		CRASH("[src] was still registed to listen in on [source] but was not found to be on their mob.")
```
Comment (Ghommie) [inline code/modules/instruments/items.dm:206]:
```suggestion
		CRASH("[src] was still registered to listen in on [source] but was not found to be on their mob.")
```

File: code/modules/instruments/songs/editor.dm [hunk 3]
```diff
@@ -1,96 +1,189 @@
...
+			data["sustain_mode_duration"] = sustain_exponential_dropoff
+			data["sustain_mode_min"] = INSTRUMENT_EXP_FALLOFF_MIN
+			data["sustain_mode_max"] = INSTRUMENT_EXP_FALLOFF_MAX
+	data["instrument_ready"] = using_instrument?.ready()
+	data["volume"] = volume
+	data["volume_dropoff_threshold"] = sustain_dropoff_volume
+	data["sustain_indefinitely"] = full_sustain_held_note
+	data["playing"] = playing
+	data["repeat"] = repeat
+	data["bpm"] = round(600 / tempo)
```
Comment (Ghommie) [inline code/modules/instruments/songs/editor.dm:33]:
```suggestion
	data["bpm"] = round(60 SECONDS / tempo)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**